File: problemas/ad-hoc/2189/2189.go
package main

import (
	"fmt"
)

func main() {
	var n, numero int

	teste := 1
	for {
		fmt.Scanf("%d", &n)

		if n == 0 {
			break
		}

		resposta := -1
		for i := 0; i < n; i++ {
			fmt.Scanf("%d", &numero)
			if i+1 == numero {
				resposta = i + 1
			}
		}

		fmt.Printf("Teste %d\n%d\n\n", teste, resposta)
		teste += 1
	}
}

File: problemas/ad-hoc/2189/2189.ml
let rec leitura teste =
  try
    Scanf.scanf "%d\n" (fun n ->
        if n != 0 then (
          let resposta = ref (-1) in
		  for i = 1 to n do
		  	Scanf.scanf "%d " (fun numero -> if numero == i then resposta := numero)
		  done;
		  Printf.printf "Teste %d\n%d\n\n" teste !resposta;
          leitura (teste + 1)))
  with End_of_file -> ()

let () = leitura 1